backend/de.metas.manufacturing/src/main/sql/postgresql/ddl/functions/PP_Product_BOM_Recursive.sql

```diff
@@ -70,23 +63,16 @@ WITH RECURSIVE bomNode AS ((SELECT ARRAY [1::integer]                      AS pa
                                                   WHERE bom.M_Product_ID = bomLineProduct.M_Product_ID
                                                     AND bom.IsActive = 'Y'
                                                     AND bom.Value = bomLineProduct.Value
-                                                  ORDER BY bom.PP_Product_BOM_ID
+                                                    AND referencedate >= bom.validfrom
+                                                  ORDER BY bom.PP_Product_BOM_ID DESC, bom.validfrom ASC
```

Comment:
```suggestion
                                                  ORDER BY bom.ValidFrom ASC, bom.PP_Product_BOM_ID DESC
```

as you demonstrated, `ORDER BY bom.PP_Product_BOM_ID DESC` does not always work